include/pybind11/cast.h

```diff
@@ -1155,15 +1155,18 @@ enable_if_t<!cast_is_temporary_value_reference<T>::value, T> cast_ref(object &&,
 // static_assert, even though if it's in dead code, so we provide a "trampoline" to pybind11::cast
 // that only does anything in cases where pybind11::cast is valid.
 template <typename T>
-enable_if_t<!cast_is_temporary_value_reference<T>::value, T> cast_safe(object &&o) {
+enable_if_t<!std::is_same<void, intrinsic_t<T>>::value
```

Comment:
@laramiel Might be a bit cleaner if use ``detail::none_of<...>` defined in detail/common.h

Reply:
Done.